File: Boca/JAVA_2/Q5/src/Livro.java
public class Livro extends Produto {
    private String autor;

    public Livro (int codigoDeBarras, String nome, double preco, String autor) {
        super(codigoDeBarras, nome, preco);
        this.autor = autor;
    }

    public String getAutor() {
        return autor;
    }

}

File: Boca/CPP/07/Livro.cpp
#include "Livro.hpp" 

Livro::Livro(int codigoBarras, const string& nome, double preco, const string& autor) : Produto(codigoBarras, nome, preco) {
    this->autor = autor;
}

void Livro::print(ostream &out) const {
    Produto::print(out);
    out << " - " << autor;
}

ostream& operator<< (ostream &out, Livro &livro) {
    livro.print(out);
    return out;
}

Livro::~Livro() {}